tefla/core/eval_metrices.py

```diff
@@ -0,0 +1,448 @@
+    except:
+        return None
+    
+def calc_npv(tn,fn):
+    """function to calculate negative predictive value
+    """
+    try:
+        npv = tn/float(tn+fn)
+        return round(npv,3)
+    except:
+        return None
+    
+def calc_fnr(fn,tp):
+    """function to calculate false negative rate
+    """
+    try:
+        fnr = fn/float(tp+fn)
+        return round(fnr,3)
+    except:
+        return None
+    
+def calc_fpr(fp,tn):
+    """function to calculate false positve rate
+    """
+    try:
+        fpr = fp/float(tn+fp)
+        return round(fpr,3)
+    except:
+        return None
+
+def calc_fdr(tp,fp):
+    """function to calculate false discovery rate
+    """
+    try:
+        fdr = fp/float(tp+fp)
+        return round(fdr,3)
+    except:
+        return None
+    
+def calc_for(fn,tn):
+    """function to calculate false positve rate
+    """
+    try:
+        fomr = fn/float(fn+tn)
+        return round(fomr,3)
+    except:
+        return None
+    
+def calc_mcc(tp,tn,fp,fn):
+    """funcrtion to calculate matthews correlation coefficient
+    """
+    try:
+        mcc = (tp*tn-fp*fn)/float(pow((tp+fp)*(tp+fn)*(tn+fp)*(tn+fn),0.5))
+        return round(mcc,3)
+    except:
+        return None
+def calc_kappa(truth,pred):
+    """funcrtion to calculate cohen's kappa coefficient
+    """
+    try:
+        kappa = metrics.cohen_kappa_score(truth,pred)
+        return round(kappa,3)
+    except:
+        return None
+    
+def calc_mae(truth,pred):
+    """function to calculate mean absolute error
+    """
+    try:
+        mae = mean([abs(truth[i]-pred[i]) for i in range(0,len(truth))])
+        return round(mae,3)
+    except:
+        return None
+    
+def calc_mse(truth,pred):
+    """function to calculate mean squared error
+    """
+    try:
+        mse = mean([pow(truth[i]-pred[i],2) for i in range(0,len(truth))])
+        return round(mse,3)
+    except:
+        return None
+def calc_rmse(truth,pred):
+    """function to calculate mean squared error
+    """
+    try:
+        rmse = pow(mean([pow(truth[i]-pred[i],2) for i in range(0,len(truth))]),0.5)
+        return round(rmse,3)
+    except:
+        return None
+def plot_conf_mat(cm,classes):
+    colors = ['#d5181c', '#fdbe62', '#ffffef', '#a6e99a', '#2a9631']
+    mapper = LinearColorMapper(palette=colors, low=cm.min(), high=cm.max())
+    p = figure(title = "confusion matrix",x_axis_label="predicted",y_axis_label="Actual",
+               x_range=[str(cls) for cls in classes], y_range=[str(cls) for cls in classes],tooltips=[("value", "@image")])
+    p.image(image=[cm],x=0,y=0,dw=len(classes),dh=len(classes), palette="Spectral11")
+    color_bar = ColorBar(color_mapper=mapper,location=(0, 0),ticker=BasicTicker(desired_num_ticks=len(colors)))
+    p.add_layout(color_bar, 'right')
+    show(p)
+
+
+
+class evaluation():
+    def __init__(self,truth_file,pred_files,eval_list,plot_list,over_all=False,ensemble_voting="soft",ensemble_weights=None):
```

Comment:
@mkulariya1 `truth_file`, `pred_files` `eval_list`, `plot_list`should be passed as arguments to `eval_*` not as in the class `__init__` (as t restrict the use for the given files only).

Reply:
resolved